### PR TITLE
Gallery and media extension updates and testing

### PR DESCRIPTION
**Pull request by aeslaughter**

### CODE REVIEW

File: python/MooseDocs/extensions/media.py [hunk 1]
```diff
@@ -74,65 +52,79 @@ def extend(self, reader, renderer):
...
-            vid = Video(flt,
-                        src=info['subcommand'],
-                        youtube=False,
-                        tex=self.settings['latex_src'],
-                        controls=self.settings['controls'],
-                        loop=self.settings['loop'],
-                        autoplay=self.settings['autoplay'],
-                        tstart=self.settings['tstart'],
-                        tstop=self.settings['tstop'])
+                                  bottom=True, img=True, **self.attributes)
```
Comment (aeslaughter) [inline python/MooseDocs/extensions/media.py:116]: @dschwen Adding the `**self.attributes` was the actual fix to the video size problem

**Comment by aeslaughter** [comment]
@dschwen I took some time to clean up the code quite a bit and update the tests

**Comment by dschwen** [comment]
How ironic that this should fail with

```
MooseDocs/test/extensions.all/media: OSError: [Errno 28] No space left on device
```

**Comment by dschwen** [comment]
I'll hold off until moosebuild provides a preview site to look at

**Comment by aeslaughter** [comment]
@idaholab/moose-ccb I would like to update our front page, what ideas do people have. Here is the version with a stand-alone gallery: https://mooseframework.inl.gov/docs/PRs/15962/site/

**Comment by dschwen** [comment]
What has been cool for a while now is a large size background video in the header...

We might also want to solicit video versions that are encoded straight from the output data without a detour via animated GIF. 

I'm also not a huge fan of autoplay. If we keep autoplay my next video will have sound :-D 
https://www.youtube.com/watch?v=KkGTB8ESMCM

**Comment by rwcarlsen** [comment]
I think it is worth keeping at least a couple of the cooler gifs/vids on the front page - you know - so people look at it and go "SCIIIEEEENNNNCEEEE!"

**Review by dschwen — APPROVED** [review]
lgtm

**Comment by friedmud** [comment]
I agree with @rwcarlsen (that keeps happening! lol)... we should keep a few on the front page so it doesn't look too stale.

As for a running news bar... that works... until it doesn't.  As sooon as we don't update it for a while it makes the project look really old and lifeless.  I'm for it, but we would need a dang good way to update it.

**Comment by friedmud** [comment]
@dschwen that sceeming marmot is incredibly disturbing!

**Comment by joshuahansel** [comment]
> As for a running news bar... that works... until it doesn't. As sooon as we don't update it for a while it makes the project look really old and lifeless. I'm for it, but we would need a dang good way to update it.

That's true. I guess my suggestion is just to pick some kind of layout that doesn't push the list of features down off the screen.

**Comment by dschwen** [comment]
Can we get this PR merged or does this require the discussion on how the front page should look like (with news ticker and the kitchen sink) to get hashed out first?

**Comment by joshuahansel** [comment]
@dschwen I'm all for a separate PR - picking out our kitchen sink might take a while.

**Comment by aeslaughter** [comment]
@dschwen You can merge, I will get back to this later in the week. I don't think I am going to add a news ticker, that would require more development than I willing to do at this time. I will add a nice movie/image and clean up the feature list.

**Pull request merged**